ERCS/erc-7855.md

```diff
@@ -0,0 +1,149 @@
+
+## Motivation
+
+Current fungible token standards lack built-in security features and gas optimization mechanisms. This proposal addresses these limitations by incorporating:
+- Native gas rebate system to reduce transaction costs
+- [EIP-712](./eip-712.md) compliant signatures for gasless approvals
+- Strong security measures against common vulnerabilities
+- Efficient state management and error handling
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Core Variables
+- `name`: Token name
+- `symbol`: Token symbol
+- `decimals`: Number of decimal places
+- `totalSupply`: Total token supply
+- `_balances`: Mapping of address to token balance
+- `_allowances`: Mapping of owner address to spender address to allowed amount
+
+### Features
+- Gas rebate system
+- [EIP-712](./eip-712.md) signature validation
+- Nonce tracking for replay protection
+
+### Functions
+
+#### View Functions
+
+**`name()`**
+- Returns the token name
+- Type: `string`
+
+**`symbol()`**
+- Returns the token symbol
+- Type: `string`
+
+**`decimals()`**
+- Returns number of decimals
+- Type: `uint8`
+
+**`totalSupply()`**
+- Returns total token supply
+- Type: `uint256`
+
+**`balanceOf(address account)`**
+- Returns token balance of specified account
+- Type: `uint256`
+
+**`allowance(address owner, address spender)`**
+- Returns remaining allowance for spender
+- Type: `uint256`
+
+#### State-Changing Functions
+
+**`transfer(address to, uint256 amount)`**
+- Transfers tokens to specified address
+- Includes gas rebate processing
+- Returns: `bool`
+
+**`approve(address spender, uint256 amount)`**
+- Approves spender to spend tokens
+- Returns: `bool`
+
+**`transferFrom(address from, address to, uint256 amount)`**
+- Transfers tokens between addresses
+- Requires sufficient allowance
+- Returns: `bool`
+
+**`permit(address owner, address spender, uint256 value, uint256 deadline, uint8 v, bytes32 r, bytes32 s)`**
+- Processes gasless approvals using [EIP-712](./eip-712.md) signatures
+- Validates signature and deadline
+- No return value
+
+### Events
+
+**`Transfer`**
+- Emitted on token transfers
+```solidity
+event Transfer(address indexed from, address indexed to, uint256 value)
+```
+
+**`Approval`**
+- Emitted on approval updates
+```solidity
+event Approval(address indexed owner, address indexed spender, uint256 value)
+```
+
+**`GasRebate`**
+- Emitted when gas rebate is processed
+```solidity
+event GasRebate(address indexed user, uint256 amount)
+```
+
+### Gas Rebate System
+- Limited to first 10,000 transactions
+- Rebate amount = 25% of base transaction cost
+- One-time rebate per user
+- Requires contract to hold ETH for rebates
```

Comment:
All of these requirements feel more like tunable parameters rather than constants defined in the standard. If I'm releasing my own token, why does it have to be fixed to 10,000 transactions?

Also, you should really be defining these in terms of the externally visible behaviour. How does the user claim their gas refund? How does the contract calculate how much to return? That kind of thing.